subsys/net/lib/openthread/platform/radio.c

```diff
@@ -183,7 +228,31 @@ void platformRadioProcess(otInstance *aInstance)
 						  NULL, result);
 			}
 		}
+	} else {
+		if (is_pending_event_set(pending_events_detect_energy)) {
+			radio_api->set_channel(radio_dev,
+					energy_detection_channel);
+
+			if (!radio_api->ed_scan(radio_dev,
```

Comment:
Something's off here, we just called this function in `otPlatRadioEnergyScan`, why do we do it again?

Reply:
pending_events_detect_energy event is called when radio_api->ed_scan fails. In such case we schedule the call for later